ui/apps/platform/src/Containers/Dashboard/PatternFly/Widgets/ComplianceLevelsByStandard.tsx

```diff
@@ -0,0 +1,180 @@
+
+// Adapted from `processData` function in the original DashboardCompliance.js code
+function processData(
+    searchFilter: SearchFilter,
+    sortDirection: SortBy,
+    data?: AggregationResult
+): ComplianceData | undefined {
+    if (!data) {
+        return undefined;
+    }
+    const { complianceStandards } = data;
+    const modifiedData = data.controls.results.map((result) => {
+        const aggregationName = result?.aggregationKeys[0]?.id;
+        const standard = complianceStandards.find((cs) => cs.id === aggregationName);
+        const { numPassing, numFailing } = result;
+        const standardQueryValue =
+            standardLabels[standard?.id] || aggregationName || 'Unrecognized standard';
+        const query = getUrlQueryStringForSearchFilter({
+            ...searchFilter,
+            Cluster: searchFilter.Cluster || '*',
+            standard: standardQueryValue,
+        });
+        const link = `${complianceBasePath}/${
+            urlEntityListTypes[standardEntityTypes.CONTROL]
+        }?${query}`;
+        const modifiedResult = {
+            name: standard?.name || aggregationName || 'Unrecognized standard',
+            passing: Math.round((numPassing / (numFailing + numPassing)) * 100) || 0,
+            link,
+        };
+        return modifiedResult;
+    });
+
+    const sorted = sortBy(modifiedData, [(datum) => datum.passing]);
+
+    if (sortDirection === 'asc') {
+        sorted.reverse();
+    }
+
+    return sorted;
+}
+
+type AggregationResult = {
+    controls: {
+        results: {
+            aggregationKeys: {
+                id: keyof typeof standardTypes;
+                scope: StandardEntityType;
+            }[];
+            numFailing: number;
+            numPassing: number;
+            numSkipped: number;
+            unit: StandardEntityType;
+        }[];
+    };
+    complianceStandards: {
+        id: string;
+        name: string;
+    }[];
+};
+
+type SortBy = 'asc' | 'desc';
+
+function ComplianceLevelsByStandard() {
+    const { isOpen: isOptionsOpen, onToggle: toggleOptionsOpen } = useSelectToggle();
+    const { searchFilter } = useURLSearch();
+    const [sortDataBy, setSortDataBy] = useState<SortBy>('asc');
+
+    const where = getRequestQueryStringForSearchFilter({
+        // We always need to include some value for Cluster, otherwise aggregation will be performed at the namespace level
+        ...searchFilter,
+        Cluster: searchFilter.Cluster || '*',
+    });
+    const variables = {
+        groupBy: [entityTypes.STANDARD],
+        where,
+    };
+    const { loading, error, data, previousData } = useQuery<AggregationResult>(
+        AGGREGATED_RESULTS_ACROSS_ENTITIES,
+        { variables }
+    );
+
+    const complianceData = useMemo(
+        () => processData(searchFilter, sortDataBy, data || previousData)?.slice(0, 6),
+        [searchFilter, sortDataBy, data, previousData]
+    );
+
+    return (
+        <WidgetCard
```

Comment:
Something to think about. Some of these `Options` dropdowns might be worth abstracting out into Higher-order Components in the future. For now, I think we only have a few so there's less of a pattern. But once we have more, maybe we can see if it's worthwhile to do

Reply:
Yeah there is definitely some repetition in these widgets, for the options and the rest of the header. Once the dust settles a little bit this is something I was looking into refactoring. The ViolationsBySeverity widget needs to be broken down as well, that was the first one and doesn't have some of the cleaner separation in the later widgets.